File: Sosyolobi.Api/DTOs/Auth/AuthResponse.cs
namespace Sosyolobi.Api.DTOs.Auth;

public class AuthResponse
{
    public string AccessToken { get; set; } = null!;
    public string RefreshToken { get; set; } = null!;
    public DateTime ExpiresAt { get; set; }
    public Guid UserId { get; set; }
    public bool IsNewUser { get; set; }
}

File: sosyolobi-mobile/lib/features/auth/domain/auth_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'auth_response.freezed.dart';
part 'auth_response.g.dart';

/// Mirrors `Sosyolobi.Api/DTOs/Auth/AuthResponse.cs`.
@freezed
abstract class AuthResponse with _$AuthResponse {
  const factory AuthResponse({
    required String accessToken,
    required String refreshToken,
    required DateTime expiresAt,
    required String userId,
    required String role,
    String? displayName,
    required bool isNewUser,
  }) = _AuthResponse;

  factory AuthResponse.fromJson(Map<String, dynamic> json) => _$AuthResponseFromJson(json);
}
